ethstats/ethstats.go

```diff
@@ -1019,12 +1015,19 @@ func (s *Service) assembleBlockStats(block *types.Block) *blockStats {
 	epochSize := s.engine.EpochSize()
 	blockRemain := epochSize - istanbul.GetNumberWithinEpoch(header.Number.Uint64(), epochSize)
 
-	// only assemble every valSetInterval blocks
-	if block != nil && block.Number().Uint64()%valSetInterval == 0 {
-		valSet = s.assembleValidatorSet(block, stateDB)
-	}
+	stateDB, _, err := s.backend.StateAndHeaderByNumberOrHash(context.Background(), rpc.BlockNumberOrHashWithHash(header.Hash(), true))
+
+	if err == nil {
+		log.Warn("Block state unavailable for reporting block stats", "hash", header.Hash(), "number", header.Number.Uint64(), "err", err)
```

Comment:
It seems like its not your day today @nategraf :D we want to log in the case of an error!

```suggestion
	if err != nil {
		log.Warn("Block state unavailable for reporting block stats", "hash", header.Hash(), "number", header.Number.Uint64(), "err", err)
	} else {
```

Reply:
🙏  I guess I knew I was tired, but someone might need to take aware my dev license, haha. This coding is reckless endangerment.

Reply:
hehe!